pkg/controller/daemon/daemon_controller.go

```diff
@@ -1259,6 +1259,18 @@ func (dsc *DaemonSetsController) simulate(newPod *v1.Pod, node *v1.Node, ds *app
 		})
 	}
 
+	// This is memory efficient copy of metav1.IsControlledBy where we don't deep copy all references.
```

Comment:
not sure we really need this comment in the code. It fits into the commit message. There is nothing wrong or special to access p.ObjectMeta.OwnerReferences directly without an accessor. This comment suggests it is uncommon.

Reply:
fixed